action/action_deserializer.go

```diff
@@ -6,15 +6,93 @@
 
 package action
 
-import "github.com/iotexproject/iotex-proto/golang/iotextypes"
+import (
+	"github.com/iotexproject/go-pkgs/crypto"
+	"github.com/iotexproject/iotex-core/config"
+	"github.com/iotexproject/iotex-proto/golang/iotextypes"
+	"github.com/pkg/errors"
+)
 
 // Deserializer de-serializes an action
 type Deserializer struct {
+	withChainID bool
 }
 
 // ActionToSealedEnvelope converts protobuf to SealedEnvelope
 func (ad *Deserializer) ActionToSealedEnvelope(pbAct *iotextypes.Action) (SealedEnvelope, error) {
```

Comment:
it can work, a different way/style to do this

the PR used `Deserializer.WithChainID(true or false). ActionToSealedEnvelope()`, i prefer this style since it is more apparent and has better readability, similar to `envelopeBuilder.SetVersion().SetGas().SetAction().Build`

Reply:
also, in the future when removing global `evmID`, it will be like 
```
Deserializer.SetEvmID(id).ActionToSealedEnvelope()
```